app/src/main/java/com/example/directory/MainActivity.kt

```diff
@@ -0,0 +1,59 @@
+package com.example.directory
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import com.example.directory.api.RetrofitHelper
+import com.example.directory.api.UserService
+import com.example.directory.databinding.ActivityMainBinding
+import com.example.directory.factory.MainViewModelFactory
+import com.example.directory.repository.UsersRepository
+import com.example.directory.viewmodels.MainViewModel
+import com.squareup.picasso.Callback
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var mainViewModel: MainViewModel
+    private lateinit var binding: ActivityMainBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityMainBinding.inflate(layoutInflater)
+
+        setContentView(R.layout.activity_main)
+
+        val userService = RetrofitHelper.getInstance().create(UserService::class.java)
+        val usersRepository = UsersRepository(userService)
```

Comment:
lets do this in the init of viewmodel. userService and userRepository will be created again and again on configuration change if kept in activity.